### ISSUE TITLE
TensorFlow Lite Error: Regular TensorFlow ops are not supported by this interpreter. Make sure you apply/link the Flex delegate before inference. TensorFlow Lite Error: Node number 0 (FlexConv3D) failed to prepare.  Fatal error: Failed to allocate memory for input tensors.

### ISSUE THREAD

**Issue by HemantMehtaM17** [tensorflow/tensorflow#44265]
Hi,

I am trying to use Activity i3D model on mobile devices. Have successfully Converted the i3D model check points to tflite. Now while loading the tflite model into the mobile application using TensorFlowLite Library. Getting the following error.

GitHub link to checkpoints : [https://github.com/deepmind/kinetics-i3d](https://github.com/deepmind/kinetics-i3d)

**TensorFlow Lite Error: Regular TensorFlow ops are not supported by this interpreter. Make sure you apply/link the Flex delegate before inference.
TensorFlow Lite Error: Node number 0 (FlexConv3D) failed to prepare.

Fatal error: Failed to allocate memory for input tensors.**

Please help me find anything I am missing.
Thanks :-)

**Comment by jvishnuvardhan** [comment]
@HemantMehtaM17 Can you please share a simple standalone to reproduce the error? Thanks!

**Comment by HemantMehtaM17** [comment]
Hi jvishnuvardhan,

Thanks for your reply. Here is the things you have asked for

I have downloaded the frozen graph from I3D git for kinetics

[https://download.01.org/opencv/public_models/032020/i3d-rgb/rgb.frozen.pb](https://download.01.org/opencv/public_models/032020/i3d-rgb/rgb.frozen.pb) 

with the help of following script have successfully converted frozen graph to tflite file to use in mobile application.


import tensorflow as tf
with tf.compat.v1.Session() as sess:
    sess.run(tf.compat.v1.global_variables_initializer())
    converter = tf.compat.v1.lite.TFLiteConverter.from_frozen_graph(
    'downloaded_frozen_activity.pb', ['Placeholder'], ['Softmax'])
   
    converter.allow_custom_ops=True
    converter.target_ops = [tf.compat.v1.lite.OpsSet.SELECT_TF_OPS]
    
     
    tflite_model = converter.convert()
    open("coverted.tflite", "wb").write(tflite_model)

with the help of this tflite file I have created a demo project in iOS. You can see the project file from below link

[https://github.com/HemantMehtaM17/i3DDemo](https://github.com/HemantMehtaM17/i3DDemo)

getting this error

**TensorFlow Lite Error: Regular TensorFlow ops are not supported by this interpreter. Make sure you apply/link the Flex delegate before inference.
TensorFlow Lite Error: Node number 0 (FlexConv3D) failed to prepare.

Fatal error: Failed to allocate memory for input tensors.**



Thanks

**Comment by jvishnuvardhan** [comment]
@HemantMehtaM17 You are using SELECT_TF_OPS during conversion. In this case, you will need to use the Interpreter which supports SELECT mode.

Please check [this comment](https://github.com/tensorflow/tensorflow/issues/43934#issuecomment-714832778) for more info on the root-cause and resource to resolve. Thanks!

**Comment by HemantMehtaM17** [comment]
@jvishnuvardhan I have used the same. But still getting the same error. You can check the pod file. I am already using the same.

**Comment by karimnosseir** [comment]
Did you add the linker flag to force linking of TF ?

go to Build Settings -> Other Linker Flags, and add:
-force_load $(SRCROOT)/Pods/TensorFlowLiteSelectTfOps/Frameworks/TensorFlowLiteSelectTfOps.framework/TensorFlowLiteSelectTfOps

See
https://www.tensorflow.org/lite/guide/ops_select#ios

**Comment by HemantMehtaM17** [comment]
Yeah I did but nothing changes.

**Comment by karimnosseir** [comment]
@yyoon Can you please check

Thanks

**Comment by yyoon** [comment]
We've recently fixed a similar issue here: https://github.com/tensorflow/tensorflow/issues/44879.
Can you try running `pod update` to update to the newest version and see if that resolves your issue?

**Comment by HemantMehtaM17** [comment]
I have updated the pod and try run the project again, getting same issue. I have already uploaded the code to GitHub please go through that.

https://github.com/HemantMehtaM17/i3DDemo

**Comment by yyoon** [comment]
What is the `TensorFlowLiteSelectTfOps` version actually installed for your project? Can you copy & paste the content of the `Podfile.lock` file here? It should point to the most recent nightly version.

I cloned your repository and confirmed that it's working fine with the latest release.
I ran the following commands from the terminal:

```sh
git clone git@github.com:HemantMehtaM17/i3DDemo.git
pod update --repo-update
```

Next, I opened the workspace with Xcode by running `open i3DDemo.xcworkspace` from the terminal, and added the linker flag mentioned above.

![image](https://user-images.githubusercontent.com/1309817/100971291-eab55780-3579-11eb-8972-aaf4b66a759b.png)

After these steps, I was able to see the following output message from the debug console, instead of the error you mentioned originally.

```
2020-12-03 15:06:59.384628+0900 i3DDemo[50920:12375299] Initialized TensorFlow Lite runtime.
2020-12-03 15:06:59.385169+0900 i3DDemo[50920:12375299] Created TensorFlow Lite delegate for select TF ops.
2020-12-03 15:06:59.385991+0900 i3DDemo[50920:12375299] TfLiteFlexDelegate delegate: 199 nodes delegated out of 199 nodes with 1 partitions.
```

Please try following these steps exactly, and let me know if this works for you.

**Comment by Waaheeda** [comment]
Hi @yyoon , Im Facing same issue on iOS 15 device , 
tried mentioned workarounds, using following pods -
 
 ` pod 'TensorFlowLiteObjC'
  pod 'libextobjc' 
  pod 'TensorFlowLiteSelectTfOps', '~> 0.0.1-nightly'`
  
  added additional linker flag to force load the select TF ops framework into project , Settings -> Other Linker Flags, 

`-force_load $(SRCROOT)/Pods/TensorFlowLiteSelectTfOps/Frameworks/TensorFlowLiteSelectTfOps.framework/TensorFlowLiteSelectTfOps`
`-u _TF_AcquireFlexDelegate`
Getting following error - 
Undefined symbol: _TfLiteTensorCo
<img width="1424" alt="Screenshot 2021-12-08 at 4 48 22 PM" src="https://user-images.githubusercontent.com/60549753/145199662-5136e74c-7629-4057-8633-b2e689c2923b.png">
py

**Comment by Waaheeda** [comment]
@yyoon  then I Updated TensorFlowLiteSelectTfOps to latest version 2.7 getting following errors 

<img width="1375" alt="Screenshot 2021-12-08 at 5 03 07 PM" src="https://user-images.githubusercontent.com/60549753/145201640-01a36229-6b88-4790-be7b-29017a7bc36f.png">

**Comment by terryheo** [comment]
Could you try to use 2.7 for `TensorFlowLiteObjC` pod ?
I think it's better to use the same version with `TensorFlowLiteSelectTfOps`.

**Comment by Waaheeda** [comment]
@terryheo thank you for quick response, it started working after I updated all pods to 2.6 version,

pod 'TensorFlowLiteObjC' , '2.6.0'
pod 'libextobjc'
pod 'TensorFlowLiteSelectTfOps' ,'2.6.0'

**Comment by terryheo** [comment]
It's good to hear. Thanks for the update.

**Comment by alarm109** [comment]
> @terryheo thank you for quick response, it started working after I updated all pods to 2.6 version,
> 
> pod 'TensorFlowLiteObjC' , '2.6.0' pod 'libextobjc' pod 'TensorFlowLiteSelectTfOps' ,'2.6.0'

Had the same issue:
```
  // Didn't work
  pod 'TensorFlowLiteSwift', '2.7.0'
  pod 'TensorFlowLiteSelectTfOps', '2.7.0'
```
```
  // Worked
  pod 'TensorFlowLiteSwift', '2.6.0'
  pod 'TensorFlowLiteSelectTfOps', '2.6.0'
```
Maybe this issue is related with arm64 mac's and 2.7.0 version?

**Comment by sseppola** [comment]
I'm on an x84_64 mac trying to run TF on my iPhone in local development, and cannot get `2.9.1` to work.

```
// does not work
pod 'TensorFlowLiteC', '2.9.1'
pod 'TensorFlowLiteSwift', '2.9.1'
pod 'TensorFlowLiteSelectTfOps', '2.9.1'
```

Reverting to `2.6.0` for all the listed pods, as mentioned by @alarm109, worked.

The latest docs do not appear to be in sync with the latest `TensorFlowLiteSelectTfOps` which uses a`xcframework`, so the path should be `$(SRCROOT)/Pods/TensorFlowLiteSelectTfOps/Frameworks/TensorFlowLiteSelectTfOps.xcframework/ios-arm64/TensorFlowLiteSelectTfOps.framework/TensorFlowLiteSelectTfOps` although I could not get that to work.

The error I get looks like a linking error:
![Screenshot 2022-08-22 at 15 06 35](https://user-images.githubusercontent.com/4065840/185928500-f63ac2f9-3dc1-4363-a1ce-2528bb8688a3.png)

@yyoon or @terryheo how can I contribute to fix this?

**Comment by yyoon** [comment]
@sseppola Could you kindly create a new issue for this? You can also tag @yishuangP in the new issue.

**Comment by yishuangP** [comment]
Hi this seems like a duplicate of https://github.com/tensorflow/tensorflow/issues/56255 and it should be fixed in the latest nightly build (version 0.0.1-nightly.20220831)
